java-frontend/src/main/java/org/sonar/java/DefaultJavaResourceLocator.java

```diff
@@ -126,19 +130,29 @@ public void scanFile(JavaFileScannerContext context) {
     javaFilesCache.scanFile(context);
     InputFile inputFile = fs.inputFile(fs.predicates().is(context.getFile()));
     org.sonar.api.resources.File currentResource = (org.sonar.api.resources.File) sensorContext.getResource(inputFile);
+    String fileKey = context.getFileKey();
     if (currentResource == null) {
-      throw new IllegalStateException("resource not found : " + context.getFileKey());
+      throw new IllegalStateException("resource not found : " + fileKey);
     }
-    resourceMapping.addResource(currentResource, context.getFileKey());
+    resourceMapping.addResource(currentResource, fileKey);
     for (Map.Entry<String, File> classIOFileEntry : javaFilesCache.getResourcesCache().entrySet()) {
       resourcesByClass.put(classIOFileEntry.getKey(), currentResource);
-      if (context.getFileKey() != null) {
-        sourceFileByClass.put(classIOFileEntry.getKey(), context.getFileKey());
+      if (fileKey != null) {
+        sourceFileByClass.put(classIOFileEntry.getKey(), fileKey);
       }
     }
     methodStartLines.putAll(javaFilesCache.getMethodStartLines());
+    String componentKey = currentResource.getEffectiveKey();
     if (javaFilesCache.hasSuppressWarningLines()) {
-      suppressWarningsFilter.addComponent(currentResource.getEffectiveKey(), javaFilesCache.getSuppressWarningLines());
+      suppressWarningsFilter.addComponent(componentKey, javaFilesCache.getSuppressWarningLines());
+    }
+    for (JavaIssueFilter javaIssueFilter : issueFilters) {
+      javaIssueFilter.addComponent(fileKey, componentKey);
```

Comment:
For what I can read you only need to set the component key, no need for an internal map on filters objects.